### PR TITLE
add cuda memory and distributed metadata

### PR DESCRIPTION
**Pull request by guotuofeng**
Implementation for https://github.com/pytorch/kineto/issues/155

### CODE REVIEW

File: torch/csrc/autograd/profiler_kineto.cpp [hunk 1]
```diff
@@ -400,6 +400,10 @@ std::unique_ptr<ProfilerResult> disableProfiler() {
 }
 
 void addMetadata(const std::string& key, const std::string& value) {
+  if (!libkineto::api().isProfilerInitialized()){
+    throw std::runtime_error("The profiler doesn't get initialized");
```
Comment (ilia-cher) [inline torch/csrc/autograd/profiler_kineto.cpp:404]: nit: "is not initialized"

**Comment by gdankel** [comment]
@guotuofeng can you add an example json or screenshot of the metadata section (when pressing M button in chrome trace)?

**Comment by guotuofeng** [comment]
> @guotuofeng can you add an example json or screenshot of the metadata section (when pressing M button in chrome trace)?

The metadata is like:

  
  "metadata": {
      "devices": "[{\"id\": 0, \"name\": \"Tesla V100-DGXS-32GB\", \"multi_processor_count\": 80, \"total_memory\": 34084028416}, {\"id\": 1, \"name\": \"Tesla V100-DGXS-32GB\", \"multi_processor_count\": 80, \"total_memory\": 34087305216}, {\"id\": 2, \"name\": \"Tesla V100-DGXS-32GB\", \"multi_processor_count\": 80, \"total_memory\": 34087305216}, {\"id\": 3, \"name\": \"Tesla V100-DGXS-32GB\", \"multi_processor_count\": 80, \"total_memory\": 34087305216}]",
    "distributed": "{\"backend\": \"nccl\", \"rank\": 0, \"world_size\": 2}"
  },
image

**Comment by ilia-cher** [comment]
(will resend)

**Pull request closed without merging**